src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -810,6 +811,7 @@ public virtual int Run(TextWriter consoleOutput, CancellationToken cancellationT
         /// <returns>A compilation that represents the original compilation with any additional, generated texts added to it.</returns>
         private protected (Compilation Compilation, GeneratorDriverTimingInfo DriverTimingInfo) RunGenerators(
             Compilation input,
+            string baseDirectory,
```

Comment:
nit: Consider asserting that `baseDirectory` is not null (to enforce the declared contract of method)